drivers/hid/hid-nintendo.c

```diff
@@ -480,23 +484,42 @@ struct joycon_ctlr {
 
 /* Helper macros for checking controller type */
 #define jc_type_is_joycon(ctlr) \
-	(ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_JOYCONL || \
-	 ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_JOYCONR || \
-	 ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_CHRGGRIP)
+	((ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_JOYCONL || \
+ 	  ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_JOYCONR || \
+ 	  ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_CHRGGRIP) && \
+ 	 !jc_type_is_nescon(ctlr))
 #define jc_type_is_procon(ctlr) \
 	(ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_PROCON)
 #define jc_type_is_chrggrip(ctlr) \
 	(ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_CHRGGRIP)
+#define jc_type_is_nescon(ctlr) \
+ 	(ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_JOYCONR && \
+ 	 (ctlr->ctlr_type == JOYCON_CTLR_TYPE_NESL || \
+ 	  ctlr->ctlr_type == JOYCON_CTLR_TYPE_NESR))
+#define jc_type_is_snescon(ctlr) \
+ 	(ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_SNESCON)
+#define jc_type_is_gencon(ctlr) \
+ 	(ctlr->hdev->product == USB_DEVICE_ID_NINTENDO_GENCON)
 
 /* Does this controller have inputs associated with left joycon? */
 #define jc_type_has_left(ctlr) \
 	(ctlr->ctlr_type == JOYCON_CTLR_TYPE_JCL || \
-	 ctlr->ctlr_type == JOYCON_CTLR_TYPE_PRO)
+ 	 ctlr->ctlr_type == JOYCON_CTLR_TYPE_PRO)
```

Comment:
This extra leading space is accidental on my part, and I will remove it in a future commit.